clientv3/balancer.go

```diff
@@ -131,6 +147,59 @@ func (b *simpleBalancer) updateAddrs(eps []string) {
 	}
 }
 
+func (b *simpleBalancer) hasPinAddr(addrs []grpc.Address) bool {
+	for _, addr := range addrs {
+		if addr.Addr == b.pinAddr {
+			return true
+		}
+	}
+	return false
+}
+
+func (b *simpleBalancer) sendLoop() {
+	defer close(b.stopc)
+	var (
+		upc   chan struct{}
+		downc chan struct{}
+		addr  string
+	)
+	for {
+		b.mu.RLock()
+		upc = b.upc
+		b.mu.RUnlock()
+		select {
```

Comment:
`Close` should `close(stopc)`; this `select` should wait on `stopc` as well as `upc` so it can terminate on `Close` without needing to mess with upc/downc